### PR TITLE
Support pushing to docker Hub

### LINKED ISSUE #271: Handle docker auth to DockerHub
We currently don't inject dockerhub credentials for images without a `docker.io` prefix. This is something that gets resolved in the docker CLI, so I think its something we need to handle here. 

https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/skaffold-users/fmnS3y3zOno/6gG6rdnaCAAJ

### PR DESCRIPTION
**Pull request by dgageot**
Fix #271

Signed-off-by: David Gageot <david@gageot.net>

### CODE REVIEW

File: pkg/skaffold/docker/auth.go [hunk 1]
```diff
@@ -27,13 +27,14 @@ import (
...
+	"github.com/docker/docker/registry"
 	"github.com/moby/moby/pkg/homedir"
 	"github.com/pkg/errors"
-	"github.com/sirupsen/logrus"
 )
 
 const (
-	configFileDir = ".docker"
+	configFileDir    = ".docker"
+	dockerHubAuthKey = "ttps://index.docker.io/v1/"
```
Comment (tuananh) [inline pkg/skaffold/docker/auth.go]: missing a `h` in the link `ttps://index.docker.io/v1/` ?

Comment (dgageot) [reply pkg/skaffold/docker/auth.go]: :-)

**Comment by dgageot** [comment]
@tuananh Thanks for your help. Should be better now

File: pkg/skaffold/docker/client.go [hunk 2]
```diff
@@ -34,23 +34,23 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-type ImageAPIClient interface {
-	client.ImageAPIClient
+type DockerAPIClient interface {
+	client.CommonAPIClient
```
Comment (r2d4) [inline pkg/skaffold/docker/client.go]: We could lock this down to ImageAPIClient + RegistryAPIClient, but I think this is fine.

**Pull request merged**